apix-impl/src/main/java/eu/xenit/apix/alfresco/comments/CommentService.java

```diff
@@ -0,0 +1,134 @@
+        commentService.updateComment(alfTargetCommentNode, "", content);
+        org.alfresco.service.cmr.repository.NodeRef alfTargetDocumentNode = apixConverter.alfresco(targetDocumentNode);
+        return toComment(alfTargetDocumentNode, alfTargetCommentNode);
+    }
+
+    @Override
+    public void deleteComment(NodeRef targetCommentNode) {
+        commentService.deleteComment(apixConverter.alfresco(targetCommentNode));
+    }
+
+    @Override
+    public boolean canCreateComment(NodeRef targetNodeRef) {
+        return permissionService.hasPermission(targetNodeRef, IPermissionService.CREATE_CHILDREN);
+    }
+
+    protected Comment toComment(org.alfresco.service.cmr.repository.NodeRef documentNode,
+            org.alfresco.service.cmr.repository.NodeRef commentNodeRef) {
+        NodeRef apixCommentNodeRef = apixConverter.apix(commentNodeRef);
+        NodeMetadata commentMetadata = nodeService.getMetadata(apixCommentNodeRef);
+        String content;
+        try {
+            content = IOUtils.toString(contentService.getContent(apixCommentNodeRef).getInputStream());
+        } catch (IOException e) {
+            String message = String.format("Encountered an IOException while handling comment %s", commentNodeRef);
+            throw new AlfrescoRuntimeException(message, e);
+        }
+        Comment response = new Comment();
+        response.setId(apixCommentNodeRef);
+        response.setContent(content);
+        List<String> property = commentMetadata.properties.get(new QName(ContentModel.PROP_TITLE.toString()));
+        if (property != null && !property.isEmpty()) {
+            response.setTitle(property.get(0));
+        }
+        property = commentMetadata.properties.get(new QName(ContentModel.PROP_CREATED.toString()));
+        if (property != null && !property.isEmpty()) {
+            response.setCreatedAt(property.get(0));
+        }
+        property = commentMetadata.properties.get(new QName(ContentModel.PROP_CREATOR.toString()));
+        if (property != null && !property.isEmpty()) {
+            response.setCreatedBy(property.get(0));
+        }
+        property = commentMetadata.properties.get(new QName(ContentModel.PROP_MODIFIED.toString()));
+        if (property != null && !property.isEmpty()) {
+            response.setModifiedAt(property.get(0));
+        }
+        property = commentMetadata.properties.get(new QName(ContentModel.PROP_MODIFIER.toString()));
+        if (property != null && !property.isEmpty()) {
+            response.setModifiedBy(property.get(0));
+        }
+        setPermissions(documentNode, commentNodeRef, response);
+        return response;
+    }
+
+    protected Comment setPermissions(org.alfresco.service.cmr.repository.NodeRef documentNode,
+            org.alfresco.service.cmr.repository.NodeRef commentNodeRef, Comment targetComment) {
+        return targetComment;
+    }
```

Comment:
This method doesn't do anything and is overrided everywhere, I think it wants to be an abstract method instead.